crates/router/src/services/kafka.rs

```diff
@@ -266,28 +270,32 @@ impl KafkaProducer {
         &self,
         attempt: &PaymentAttempt,
         old_attempt: Option<PaymentAttempt>,
+        tenant_id: TenantID,
     ) -> MQResult<()> {
         if let Some(negative_event) = old_attempt {
-            self.log_event(&KafkaEvent::old(&KafkaPaymentAttempt::from_storage(
-                &negative_event,
-            )))
+            self.log_event(&KafkaEvent::old(
+                &KafkaPaymentAttempt::from_storage(&negative_event),
+                tenant_id,
+            ))
             .attach_printable_lazy(|| {
                 format!("Failed to add negative attempt event {negative_event:?}")
             })?;
         };
-        self.log_event(&KafkaEvent::new(&KafkaPaymentAttempt::from_storage(
-            attempt,
-        )))
+        self.log_event(&KafkaEvent::new(
+            &KafkaPaymentAttempt::from_storage(attempt),
+            tenant_id,
+        ))
         .attach_printable_lazy(|| format!("Failed to add positive attempt event {attempt:?}"))
     }
 
     pub async fn log_payment_attempt_delete(
         &self,
         delete_old_attempt: &PaymentAttempt,
     ) -> MQResult<()> {
-        self.log_event(&KafkaEvent::old(&KafkaPaymentAttempt::from_storage(
-            delete_old_attempt,
-        )))
+        self.log_event(&KafkaEvent::old(
+            &KafkaPaymentAttempt::from_storage(delete_old_attempt),
+            TenantID("default".to_string()),
+        ))
```

Comment:
Can we follow the same pattern as we did for `log_payment_attempt` function for every other usage as well?

there shouldn't be any new initializations of `TenantID` in this PR, since it is already defined in KafkaStore which is what we'll be using instead

Reply:
Yes, I this commit solves this issue,
let me know if further changes needed. @lsampras